### ISSUE TITLE
vagrant up controlplane fails on Windows machine

### ISSUE THREAD

**Issue by jsturtevant** [kubernetes-sigs/sig-windows-dev-tools#110]
```
    controlplane: Removing insecure key from the guest if it's present...
    controlplane: Key inserted! Disconnecting and reconnecting using new SSH key...
==> controlplane: Machine booted and ready!
==> controlplane: Checking for guest additions in VM...
==> controlplane: Setting hostname...
==> controlplane: Configuring and enabling network interfaces...
==> controlplane: Mounting shared folders...
    controlplane: /var/sync/linux => C:/Users/jsturtevant/sig-windows-dev-tools/sync/linux
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=1000,gid=1000,_netdev var_sync_linux /var/sync/linux

The error output from the command was:

: Invalid argument

```

lowered cpu/memory variables but otherwise left everything untouched:

```
linux_ram: 2048
linux_cpus: 2
windows_ram: 4096
windows_cpus: 2
```

**Comment by jsturtevant** [comment]
going to try: 

vagrant plugin install vagrant-vbguest

**Comment by jsturtevant** [comment]
That seems to have worked, I got further:

```
controlplane: /tmp/vagrant-shell: line 131: /var/sync/shared/kubeadm.yaml: No such file or directory                           
The SSH command responded with a non-zero exit status. Vagrant                                                                     
assumes that this means the command failed. The output for this command                                                           
 should be in the log above. Please read the output to determine what                                                              
 went wrong.                             
```

**Comment by jayunit100** [comment]
sounds like right now, we need to own the story for users to use windows from beggining to end.  as of now only @FriedrichWilken has done this succesfully, and in that case he used hyperv....

**Comment by FriedrichWilken** [comment]
@jsturtevant you don't have make on Windows, so I assume you only run `vagrant up`, right?

**Comment by jsturtevant** [comment]
yes. regarding the first error, I was running on windows without make, so I ran the `vargrant up` directly and missed the part in the makefile https://github.com/kubernetes-sigs/sig-windows-dev-tools/blob/9b8f4452f009c4330c817f05f8886e0be2b53624/makefile#L38

Some of the instructions in the read me tell you to install plugins others don't. Since Windows won't always have make, we should find a way run these scripts cross platform

**Comment by lippertmarkus** [comment]
I just published a Vagrant setup which works on Windows here: https://github.com/lippertmarkus/vagrant-k8s-win-hostprocess
I originally started trying to extend your existing work but couldn't get it running. It uses Hyper-V, plain Ubuntu/Windows Server 2022 boxes as well as HostProcess Pods to deploy Calico and kube-proxy to Windows. I tried to make it as simple as possible and provide a walkthrough in README 

Although not integrated, I hope that helps you anyway with making progress to support running the Dev Environment on Windows machines and adding HostProcess and Windows Server 2022 support.
